python/cuml/neighbors/kneighbors_classifier.pyx

```diff
@@ -255,14 +249,11 @@ class KNeighborsClassifier(NearestNeighbors):
         )
 
         self.handle.sync()
-        if isinstance(X, np.ndarray):
-            return np.array(classes, dtype=np.int32)
-        elif isinstance(X, cudf.DataFrame):
-            if classes.ndim == 1:
-                classes = classes.reshape(classes.shape[0], 1)
-            return cudf.DataFrame.from_gpu_matrix(classes)
-        else:
-            return classes
+
+        del knn_indices
```

Comment:
I'm not sure we need an explicit delete here since the reference count should drop to 0 after this function returns. Tagging @dantegd for thoughts.